### PR TITLE
drop support for python 3.6

### LINKED ISSUE #4688: drop python 3.6 support
NEP 29 states that as of Jun 23, 2020 we can drop python 3.6 support. `pandas` will do so in their [upcoming `1.2` release](https://github.com/pandas-dev/pandas/blob/master/doc/source/whatsnew/v1.2.0.rst#increased-minimum-version-for-python), so we might eventually be forced to do that, too.

### PR DESCRIPTION
**Pull request by keewis**
Following NEP-29, our dependencies have started to drop active support for python 3.6. I tried removing all mentions of py36, but I might have missed something.

Last time I checked support for python 3.9 was blocked by some dependencies, so adding that will have to wait a bit longer.

 - [x] Closes #4688
 - [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

### CODE REVIEW

**Comment by keewis** [comment]
it seems we have to update the minimum setuptools version for this, but I'm not sure which to choose. `39.2` is the last release before py37 was released, but I can't get conda (using `conda-forge`) to install anything below `40.4` into a environment with `python=3.7.9`

**Comment by mathause** [comment]
https://github.com/pydata/xarray/blob/master/doc/installing.rst will need to be adjusted

File: doc/installing.rst [hunk 1]
```diff
@@ -6,7 +6,7 @@ Installation
 Required dependencies
 ---------------------
 
-- Python (3.6 or later)
+- Python (3.7 or later)
 - setuptools (38.4 or later)
```
Comment (keewis) [inline doc/installing.rst:10]:
```suggestion
- setuptools (40.4 or later)
```

**Comment by keewis** [comment]
I will also change the `whats-new.rst` entry. @mathause, was there anything missing other than the `setuptools` version?

**Comment by mathause** [comment]
I think thats all.

**Comment by jhamman** [comment]
LGTM. As I understand it, we don't really have a choice on the setuptools version bump. Can we temporarily disable the check on setuptools so the CI passes?

**Comment by keewis** [comment]
there's a `POLICY_OVERRIDE` for `setuptools`, so I updated that.

Reading #4295 again, I'm not sure what to do about this: we don't really *need* to require `setuptools>=40.4` because of features / bug fixes, it's just that we can't have our CI check that that version doesn't break anything (I don't expect using the `pip` section works). I think I will update the changelog entry to reflect that, and not update the `install_requires` / `setup_requires` entries for `setuptools`.

We will only be able to fully get rid of this issue once we drop support for python 3.7 and are able to switch to the stdlib `importlib.*` libraries (and the external `entrypoints` library).

**Comment by mathause** [comment]
Do you need to update setuptools in setup.cfg?

https://github.com/pydata/xarray/blob/31d540f9d668fc5f8c1c92165f950c568778db01/setup.cfg#L79

**Comment by keewis** [comment]
> Do you need to update setuptools in setup.cfg?

:+1: we should probably keep that in sync with the environment files.

@shoyer, any opinion about the bump of `setuptools` beyond the maximum version available when py37 was released (see https://github.com/pydata/xarray/pull/4720#issuecomment-749221358)?

Otherwise this should be ready for merging.

**Comment by jhamman** [comment]
My opinion on the bump of `setuptools` is that we don't really have a choice. If 40.4 is the minimum version that works, that's the minimum version that works. We should really just be treating the minimum version policy as a guide in cases like this. In this case, we make an exception because of incompatibility outside our control.

**Comment by jhamman** [comment]
This is blocking #4730. Any objections to merging this today?

**Review by dcherian — APPROVED** [review]

**Review by jhamman — APPROVED** [review]

**Pull request merged**